### PR TITLE
feat(components/forms): move file attachment url placeholder text to hint text

### PR DESCRIPTION
**Pull request by Blackbaud-CoreyArcher**
[AB#2833965](https://dev.azure.com/blackbaud/f565481a-7bc9-4083-95d5-4f953da6d499/_workitems/edit/2833965)

### CODE REVIEW

**Comment by blackbaud-sky-build-user** [comment]
[Storybook preview](https://blackbaud.github.io/skyux-pr-preview/2113/storybook/)

Component Storybooks:

- [action-bars-storybook](https://blackbaud.github.io/skyux-pr-preview/2113/storybooks/action-bars-storybook/)
- [ag-grid-storybook](https://blackbaud.github.io/skyux-pr-preview/2113/storybooks/ag-grid-storybook/)
- [angular-tree-component-storybook](https://blackbaud.github.io/skyux-pr-preview/2113/storybooks/angular-tree-component-storybook/)
- [avatar-storybook](https://blackbaud.github.io/skyux-pr-preview/2113/storybooks/avatar-storybook/)
- [colorpicker-storybook](https://blackbaud.github.io/skyux-pr-preview/2113/storybooks/colorpicker-storybook/)
- [data-manager-storybook](https://blackbaud.github.io/skyux-pr-preview/2113/storybooks/data-manager-storybook/)
- [datetime-storybook](https://blackbaud.github.io/skyux-pr-preview/2113/storybooks/datetime-storybook/)
- [errors-storybook](https://blackbaud.github.io/skyux-pr-preview/2113/storybooks/errors-storybook/)
- [forms-storybook](https://blackbaud.github.io/skyux-pr-preview/2113/storybooks/forms-storybook/)
- [inline-form-storybook](https://blackbaud.github.io/skyux-pr-preview/2113/storybooks/inline-form-storybook/)
- [layout-storybook](https://blackbaud.github.io/skyux-pr-preview/2113/storybooks/layout-storybook/)
- [lists-storybook](https://blackbaud.github.io/skyux-pr-preview/2113/storybooks/lists-storybook/)
- [lookup-storybook](https://blackbaud.github.io/skyux-pr-preview/2113/storybooks/lookup-storybook/)
- [pages-storybook](https://blackbaud.github.io/skyux-pr-preview/2113/storybooks/pages-storybook/)
- [phone-field-storybook](https://blackbaud.github.io/skyux-pr-preview/2113/storybooks/phone-field-storybook/)
- [split-view-storybook](https://blackbaud.github.io/skyux-pr-preview/2113/storybooks/split-view-storybook/)
- [tabs-storybook](https://blackbaud.github.io/skyux-pr-preview/2113/storybooks/tabs-storybook/)
- [text-editor-storybook](https://blackbaud.github.io/skyux-pr-preview/2113/storybooks/text-editor-storybook/)
- [theme-storybook](https://blackbaud.github.io/skyux-pr-preview/2113/storybooks/theme-storybook/)

Apps:

- [code-examples](https://blackbaud.github.io/skyux-pr-preview/2113/code-examples/)
- [dep-graph](https://blackbaud.github.io/skyux-pr-preview/2113/dep-graph/)
- [integration](https://blackbaud.github.io/skyux-pr-preview/2113/integration/)
- [playground](https://blackbaud.github.io/skyux-pr-preview/2113/playground/)

File: libs/components/forms/src/lib/modules/file-attachment/file-drop.component.html [hunk 1]
```diff
@@ -89,23 +89,30 @@
...
             (keyup)="addLinkEnter($event)"
           />
         </div>
+        <div
+          skyId
+          #hintText="skyId"
+          class="sky-font-deemphasized sky-file-drop-hint-text sky-margin-stacked-sm"
+        >
+          {{
+            'skyux_file_attachment_file_upload_link_placeholder'
```
Comment (Blackbaud-SteveBrush) [inline libs/components/forms/src/lib/modules/file-attachment/file-drop.component.html:112]: The screenshots in the story have the hint text being, "Start with http:// or https://"

File: libs/components/forms/src/lib/modules/file-attachment/file-drop.component.html [hunk 2]
```diff
@@ -89,23 +89,30 @@
...
             class="sky-form-control"
             [attr.aria-label]="
               linkUploadAriaLabel ||
               ('skyux_file_attachment_file_upload_link_label' | skyLibResources)
             "
-            [attr.placeholder]="
-              'skyux_file_attachment_file_upload_link_placeholder'
-                | skyLibResources
-            "
+            [attr.aria-describedby]="hintText.id"
```
Comment (Blackbaud-SteveBrush) [inline libs/components/forms/src/lib/modules/file-attachment/file-drop.component.html:100]: I'd add a unit test confirming this change.

File: libs/components/forms/src/lib/modules/file-attachment/file-drop.component.html [hunk 3]
```diff
@@ -89,23 +89,30 @@
...
+        <div
+          skyId
+          #hintText="skyId"
+          class="sky-font-deemphasized sky-file-drop-hint-text sky-margin-stacked-sm"
+        >
+          {{
+            'skyux_file_attachment_file_upload_link_placeholder'
+              | skyLibResources
+          }}
+        </div>
```
Comment (Blackbaud-SteveBrush) [inline libs/components/forms/src/lib/modules/file-attachment/file-drop.component.html:115]: From the story, it appears that there's also hint text underneath the "Drag a file here" section. Do we need to account for that in this PR?

File: libs/components/forms/src/lib/modules/file-attachment/file-drop.component.html (hunk 1 above)
Comment (Blackbaud-CoreyArcher) [reply libs/components/forms/src/lib/modules/file-attachment/file-drop.component.html:112]: Changed

File: libs/components/forms/src/lib/modules/file-attachment/file-drop.component.html (hunk 2 above)
Comment (Blackbaud-CoreyArcher) [reply libs/components/forms/src/lib/modules/file-attachment/file-drop.component.html:100]: Added

File: libs/components/forms/src/lib/modules/file-attachment/file-drop.component.html (hunk 3 above)
Comment (Blackbaud-CoreyArcher) [reply libs/components/forms/src/lib/modules/file-attachment/file-drop.component.html:115]: That hint text is a different story

**Review by Blackbaud-SteveBrush — APPROVED** [review]

**Pull request merged**